Error retrieving quarkus.http.auth.form.cookie-name

After Quarkus refactoring in quarkusio/quarkus#45769 the `quarkus.http.auth.form.cookie-name` setting is available only at runtime.
`HillaSecurityRecorder.setupFormAuthenticationMechanism` should be changed to avoid getting the value at build time, postponing the lookup in the supplier instance or in the constructor of `HillaFormAuthenticationMechanism`.

```
  Caused by: java.util.NoSuchElementException: SRCFG00014: The config property quarkus.http.auth.form.cookie-name is required but it could not be found in any config source
  	at io.smallrye.config.SmallRyeConfig.convertValue(SmallRyeConfig.java:507)
  	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:452)
  	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:431)
  	at com.github.mcollovati.quarkus.hilla.security.HillaSecurityRecorder.setupFormAuthenticationMechanism(HillaSecurityRecorder.java:34)
  	at io.quarkus.runner.recorded.QuarkusHillaSecurityProcessor$registerHillaFormAuthenticationMechanism1975585193.deploy_0(Unknown Source)
  	at io.quarkus.runner.recorded.QuarkusHillaSecurityProcessor$registerHillaFormAuthenticationMechanism1975585193.deploy(Unknown Source)
  	... 8 more
```

--- a/commons/runtime/src/main/java/com/github/mcollovati/quarkus/hilla/security/HillaSecurityRecorder.java
+++ b/commons/runtime/src/main/java/com/github/mcollovati/quarkus/hilla/security/HillaSecurityRecorder.java
@@ -32,17 +32,17 @@ import com.github.mcollovati.quarkus.hilla.QuarkusHillaExtension;
 public class HillaSecurityRecorder {
 
     public Supplier<HillaFormAuthenticationMechanism> setupFormAuthenticationMechanism() {
-        Config config = ConfigProvider.getConfig();
-        VaadinSecurityConfig securityConfig =
-                config.unwrap(SmallRyeConfig.class).getConfigMapping(VaadinSecurityConfig.class);
-        var authConfig = new HillaFormAuthenticationMechanism.Config(
-                config.getValue("quarkus.http.auth.form.cookie-name", String.class),
-                config.getOptionalValue("quarkus.http.auth.form.landing-page", String.class)
-                        .orElse("/"),
-                securityConfig.logoutPath(),
-                securityConfig.postLogoutRedirectUri().orElse(null),
-                securityConfig.logoutInvalidateSession());
         return () -> {
+            Config config = ConfigProvider.getConfig();
+            VaadinSecurityConfig securityConfig =
+                    config.unwrap(SmallRyeConfig.class).getConfigMapping(VaadinSecurityConfig.class);
+            var authConfig = new HillaFormAuthenticationMechanism.Config(
+                    config.getValue("quarkus.http.auth.form.cookie-name", String.class),
+                    config.getOptionalValue("quarkus.http.auth.form.landing-page", String.class)
+                            .orElse("/"),
+                    securityConfig.logoutPath(),
+                    securityConfig.postLogoutRedirectUri().orElse(null),
+                    securityConfig.logoutInvalidateSession());
             FormAuthenticationMechanism delegate =
                     Arc.container().instance(FormAuthenticationMechanism.class).get();
             return new HillaFormAuthenticationMechanism(delegate, authConfig);

